### ISSUE TITLE
variable is declared but never used with --noUnusedLocals and spread (React)

### ISSUE THREAD

**Issue by iffy** [Microsoft/TypeScript#15802]
**TypeScript Version:**  2.3.1

**Code**

```tsx
// hello.tsx
import * as React from 'react';
export class SelectInput extends React.Component<any,any> {
  render() {
    let cls = 'hello';
    return (<select {...this.props} className={cls}>{this.props.children}</select>);
  }
}
```

`package.json`:

```json
{
  "dependencies": {
    "@types/react": "^15.0.24",
    "react": "^15.5.4",
    "typescript": "2.3.1"
  }
}
```

**Expected behavior:**

I expect the compiler to think that `cls` was used.

**Actual behavior:**

The compiler emits this error:

```
$ tsc --jsx react --noUnusedLocals hello.tsx 
hello.tsx(5,9): error TS6133: 'cls' is declared but never used.
```

**Comment by iffy** [comment]
I should mention, I'm not sure if it's the same or different than:

- https://github.com/Microsoft/TypeScript/issues/12766
- https://github.com/Microsoft/TypeScript/issues/11324

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/15478, should be fixed in TS 2.3.3.

The fix should be in `typescript@next` already, give it a try and let us know if you are still running into issues.
